src/app/InteractionModelEngine.cpp

```diff
@@ -1111,21 +1119,25 @@ bool InteractionModelEngine::EnsureResourceForSubscription(FabricIndex aFabricIn
     // oldest) subscriptions from the current fabric until we have enough resource for the new subscription.
     didEvictHandler = true;
     while ((usedAttributePaths + aRequestedAttributePathCount > attributePathCap ||
-            usedEventPaths + aRequestedEventPathCount > eventPathCap || usedReadHandlers >= readHandlerCap) &&
+            usedEventPaths + aRequestedEventPathCount > eventPathCap || 
+            usedReadHandlers >= readHandlerCap) &&
            // Avoid infinity loop
            didEvictHandler)
     {
         didEvictHandler = evictAndUpdateResourceUsage(aFabricIndex, true);
     }
 
     // If didEvictHandler is false, means the loop above evicted all subscriptions from the current fabric but we still don't have
-    // enough resources for the new subscription, this should never happen.
+    // enough resources for the new subscription, this should rarely happen. Only if there are no subscriptions to evict.
+    // This happens if other fabrics are at or under quote and aFabricIndex has no subscriptions at all to convict,
+    // but the remaining read hanlders are used up by normal reads.  
     // This is safe as long as we have rejected subscriptions without a fabric associated (with a PASE session) before.
```

Comment:
I assume "This is safe" refers to the `VerifyOrDieWithMsg` that comes next. If I understand the problem correctly, this is not always the case  for us when there is a large number of subscriptions.

Reply:
right, that was the problem.